avocado/utils/cpu.py

```diff
@@ -195,3 +197,59 @@ def offline(cpu):
     if _get_cpu_status(cpu):
         return 1
     return 0
+
+
+def get_cpuidle_state():
+    """
+    Get current cpu idle values
+
+    :return: Dict of cpuidle states values for all cpus
+    :rtype: Dict of dicts
+    """
+    cpus_list = cpu_online_list()
+    states = range(len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+    cpu_idlestate = {}
+    for cpu in cpus_list:
+        cpu_idlestate[cpu] = {}
+        for state_no in states:
+            state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (cpu, state_no)
+            try:
+                cpu_idlestate[cpu][state_no] = int(open(state_file).read())
+            except IOError as err:
+                logging.warning("Failed to read idle state on cpu %s "
+                                "for state %s:\n%s", cpu, state_no, err)
+    return cpu_idlestate
+
+
+def set_cpuidle_state(state_number="all", disable=1, oldstate=None):
```

Comment:
welll the "oldstate" doesn't necessary have to be old. How about "setstate" or just "state" as it basically uses the "state" and applies it.

Reply:
Sure..